tests/tests.cc

```diff
@@ -29,6 +29,89 @@ int TestRun::printf(char const *fmt, ...) {
   return result;
 }
 
+extern "C" void *__capability map_load_only_impl(TestRun *self) {
+  size_t length = 42 * 4096;
+  void *ptr = mmap(nullptr, length, PROT_READ, MAP_ANONYMOUS, -1, 0);
+#ifdef __CHERI_PURE_CAPABILITY__
+  void *__capability cap = cheri_perms_and(ptr, CHERI_PERM_LOAD);
+#else
+  auto addr = reinterpret_cast<ptraddr_t>(ptr);
+  void *__capability cap = cheri_address_set(cheri_ddc_get(), addr);
+  cap = cheri_bounds_set_exact(cap, length);
+  cap = cheri_perms_and(cap, CHERI_PERM_LOAD);
+#endif
+  if (self->options().verbose()) {
+    self->printf("map_load_only() -> %#lp\n", ptr);
+  }
+  return cap;
+}
+
+// Get a capability with load, but not load-capability permissions, and make
+// some effort to clean up temporaries, etc.
+//
+// This is not expected to be reliable enough for security purposes, but is good
+// enough for tests.
+__attribute((naked)) void *__capability TestRun::map_load_only() {
+  // AAPCS64(-cap):
+  //    c0/x0: `this`
+  // The result is returned in c0.
+  // clang-format off
+  asm(
+#ifdef __CHERI_PURE_CAPABILITY__
+      "str  clr, [csp, #-16]!\n"
+#else
+      "stp  lr, xzr, [sp, #-16]!\n"
+#endif
+      "bl   map_load_only_impl\n"
+      // Scrub all caller-saved capability registers except:
+      //  - c0, which holds the result,
+      //  - c17, which we'll use to scrub the stack,
+      //  - and clr, which we're going to restore anyway.
+      "mov  x1, #0\n"
```

Comment:
> Is that intentional?

Yes. There's no move-immediate instruction that writes C registers, but since they're zero-extended, it has the same effect.

> is `mov xn, #0` better than `mov xn, xzr`

A quick look through a few software optimisation guides (like [this one](https://developer.arm.com/documentation/PJDOC-466751330-9707/r4p1/?lang=en)) suggests that it makes no difference.